### PR TITLE
Point feature for Pluribus Networks netvisor

### PR DESCRIPTION
**Pull request by rajaspachipulusu17**
##### SUMMARY
Point feature for Pluribus Networks Netvisor switches 

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

pn_access_list

##### ADDITIONAL INFORMATION

```paste below
ansible 2.4.0.0
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
```

### CODE REVIEW

**Comment by Akasurde** [comment]
@rajaspachipulusu17 Thanks for the PR. Any reason that we are removing other modules while adding new modules?

**Comment by rajaspachipulusu17** [comment]
> @rajaspachipulusu17 Thanks for the PR. Any reason that we are removing other modules while adding new modules?

Those are old modules. It has compatibility issues.

**Comment by Akasurde** [comment]
@rajaspachipulusu17 In order to remove / deprecate a module, you need to follow [module deprecation workflow](https://docs.ansible.com/ansible/latest/dev_guide/module_lifecycle.html#deprecating-modules).

File: lib/ansible/module_utils/network/netvisor/pn_nvos.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#
```
Comment (Akasurde) [inline lib/ansible/module_utils/network/netvisor/pn_nvos.py]: `module_utils` generally covered under BSD License, so 

Replace line 1-16 with
```
# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
```

File: lib/ansible/modules/network/netvisor/pn_access_list.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/python
+#
```
Comment (Akasurde) [inline lib/ansible/modules/network/netvisor/pn_access_list.py]: Replace line 2 to 17 with 

```
# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)


from __future__ import absolute_import, division, print_function
__metaclass__ = type
```

Comment (Akasurde) [inline lib/ansible/modules/network/netvisor/pn_access_list.py]: Add Copyright like 

```
# Copyright: (c) 2018, Pluribus Networks
```

File: lib/ansible/modules/network/netvisor/pn_access_list.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+
+DOCUMENTATION = """
+---
+module: pn_access_list
+author: "Pluribus Networks (devops@pluribusnetworks.com)"
+version_added: "2.8"
+short_description: CLI command to create/delete access-list.
+description:
+  - create and delete an access list
```
Comment (Akasurde) [inline lib/ansible/modules/network/netvisor/pn_access_list.py]:
```suggestion
  - This module can be used to create and delete an access list.
```

File: lib/ansible/modules/network/netvisor/pn_access_list.py [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = """
+---
+module: pn_access_list
+author: "Pluribus Networks (devops@pluribusnetworks.com)"
+version_added: "2.8"
+short_description: CLI command to create/delete access-list.
```
Comment (Akasurde) [inline lib/ansible/modules/network/netvisor/pn_access_list.py]:
```suggestion
short_description: CLI command to create/delete access-list
```

File: lib/ansible/modules/network/netvisor/pn_access_list.py [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+    type: str
+  state:
+    description:
+      - State the action to perform. Use 'present' to create access-list and
+        'absent' to delete access-list.
+    required: True
+    choices: [ "present", "absent"]
+  pn_name:
+    description:
+      - Access List Name
```
Comment (Akasurde) [inline lib/ansible/modules/network/netvisor/pn_access_list.py]:
```suggestion
      - Access List Name.
```

File: lib/ansible/modules/network/netvisor/pn_access_list.py [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+    required: false
+    type: str
+  pn_scope:
+    description:
+      - scope - local or fabric
```
Comment (Akasurde) [inline lib/ansible/modules/network/netvisor/pn_access_list.py]:
```suggestion
      - 'scope. Available valid values - local or fabric.'
```

File: lib/ansible/modules/network/netvisor/pn_access_list.py [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+    # Response in JSON format
+    if result != 0:
+        module.exit_json(
+            command=' '.join(cmd).replace(remove_cmd, ''),
+            stderr=err.strip(),
+            msg="%s operation failed" % command,
+            changed=False
+        )
+
+    if out:
```
Comment (Akasurde) [inline lib/ansible/modules/network/netvisor/pn_access_list.py:122]: This block can be simplified as - 

```Python
    results = dict(
        command=' '.join(cmd).replace(remove_cmd, ''),
        msg="%s operation completed" % command,
        changed=True
    )

    if out:
        results['stdout'] = out.strip()
    
    module.exit_json(**results)
    
```

**Review by Akasurde — CHANGES_REQUESTED** [review]

**Comment by rajaspachipulusu17** [comment]
Can i know what is the procedure to merge this commit. I have made all the requested changes.

**Comment by Akasurde** [comment]
@rajaspachipulusu17 First we need to merge this PR https://github.com/ansible/ansible/pull/48791 then we can start working on this.

**Comment by rajaspachipulusu17** [comment]
Hi Team,Should I change any thing to merge #48791? There is no comments yet it's been 4 days now.Thanks,RajasOn 19-Nov-2018 4:52 PM, Abhijeet Kasurde <notifications@github.com> wrote:@rajaspachipulusu17 First we need to merge this PR #48791 then we can start working on this.

**Comment by amitsi** [comment]
ship it

**Comment by amitsi** [comment]
shipit

**Comment by rajaspachipulusu17** [comment]
Creating another PR as ansibot added stale_cl and stale_review state.

**Comment by rajaspachipulusu17** [comment]
PR https://github.com/ansible/ansible/pull/49176

**Pull request closed without merging**